applications/Preprocessor/lib/MeshSource.h

```diff
@@ -159,8 +163,17 @@ class MeshSource2 {
          */
         std::string zoneName;
 
+        /**
+         * Dimension of the Element
+         */
+        size_t dimension;
+        /**
+         * Index of the Element
+         */
+        size_t id;
+
         bool operator==(const Element& o) const {
-            return coordinateIds == o.coordinateIds;
+            return coordinateIds == o.coordinateIds && zoneName == o.zoneName;
```

Comment:
Why is this needed? To me it seems quite strange if adding/removing zone names would make elements different.